packages/react-components/react-drawer/src/components/DrawerHeaderTitle/useDrawerHeaderTitle.ts

```diff
@@ -51,7 +34,14 @@ export const useDrawerHeaderTitle_unstable = (
       }),
       { elementType: 'div' },
     ),
-    heading: headingProps,
+    heading: slot.optional(heading, {
```

Comment:
curious what's the benefit of inlining this vs having it encapsulated in self explanatory private hook ? 

if there is none this feels rather than stylistic change not worth adding atm ?

Reply:
> curious what's the benefit of inlining this vs having it encapsulated in self explanatory private hook ?

- to align with other components and keep consistent design
- complexity embedded into that hook does not worth encapsulation (it had sense before when the implementation was different)
- it's not self explanatory, it's confusing as it's named `useHeadingProps()` while it returns a `Slot` type

Reply:
do we have lint rules and style guide for this ? if not we rather start creating one. as catching this in every PR might be not very productive.

Reply:
> do we have lint rules and style guide for this ? if not can we start creating one ? as catching this in every PR might 
be not very productive.

Agree 👍 

No, we don't have lint rules, for this scenario I am struggling to define what to lint (valid/invalid case to be general enough). We also don't have a style guide as it's creation have been postponed until the composition pattern become stable. Should I create an issue about creating style guide?

Reply:
> Should I create an issue about creating style guide?

sounds good ! ty

Reply:
@Hotell https://github.com/microsoft/fluentui/issues/29779